pybadges/trend.py

```diff
@@ -0,0 +1,59 @@
+from typing import Optional, List, Tuple
+
+import drawSvg as draw
+import itertools
+import numpy as np
+
+import pybadges
+
+HEIGHT = 13
+WIDTH = 110
+X_OFFSET = 7
+Y_OFFSET = 1
+
+
+def normalize(arr: np.ndarray) -> np.ndarray:
+    max_arr = np.max(arr)
+    if max_arr != 0:
+        arr /= max_arr
+    return arr
+
+
+def repeat(samples: List[int], n: int) -> List[int]:
+    """Repeats a value n times in an array.
+
+    Args:
+        samples: The list of all elements to be repeated.
+        n: Number of times to repeat each element in samples.
+    """
+    return list(
+        itertools.chain.from_iterable(
+            itertools.repeat(sample, n) for sample in samples))
+
+
+def fit_data(samples: List[int]) -> Tuple[List[int], List[int]]:
+    y = list(
+        itertools.chain.from_iterable(
+            itertools.repeat(sample, 10) for sample in samples))
+    xp = np.arange(len(y))
+    yp = normalize(np.poly1d(np.polyfit(xp, y, 15))(xp))
```

Comment:
So the bottom is always zero?

Reply:
Ahm, I don't quite get the question.

If all the values in `yp` are equal, then we'd have an array of all 1's after normalization. Similarly for all zeroes, we'd have a flat line.

Polyfit might return negative numbers sometime (`-1` at most). But, since we only scale up positive numbers (`yp[yp > 0] *= ...`), and use an offset of `-1` when defining the origin, it shouldn't be a problem.